### ISSUE TITLE
Files are read from file system instead vinyl

### ISSUE THREAD

**Issue by marcioj** [shama/webpack-stream#72]
I'm using `webpack-stream` with [gulp-preprocess](https://github.com/jas/gulp-preprocess), and in some file I have a `someVar = /* @echo CONFIG */` which is processed by `gulp-preprocess` and become `someVar = { foo: "bar" }` in order to expose the configuration to the client.
But I noticed that `webpack-stream` is just reading the contents from the original file in the file system instead the processed, so babel-loader gives syntax error, since `someVar = /* @echo CONFIG */` is not valid javascript.

I created this [sample repo](https://github.com/marcioj/gulp-webpack-issue) to easily reproduce the problem.

**Comment by rstoenescu** [comment]
Same problem here. I am preprocessing the original files and then stream them to `webpack-stream`. Would really like to see this fixed as otherwise we can't really leverage streams (for example in Gulp) at their full potential.

**Comment by marcioj** [comment]
In my case I get around this issue replacing the gulp-preprocess plugin with the webpack [DefinePlugin](https://github.com/webpack/docs/wiki/list-of-plugins#dependency-injection)

**Comment by webcaetano** [comment]
Had same problem as my countryman @marcioj . With gulp-preprocess
The plugin DefinePlugin works.

**Comment by ldskywalker** [comment]
same problem + 1.  The plugin does work as a workaround, however I still prefer this bug being fixed.

**Comment by dcholth** [comment]
Same issue.  Looking into DefinePlugin now.  Thanks for finding that!

**Comment by YodasWs** [comment]
Yes, I, too, want to process my files and pass the results into `gulp-webpack`. However, `gulp-webpack` simply takes the file names from the stream to pass into `webpack` to load the files and not use the vinyl/stream being passed into `gulp-webpack`&hellip;

in `gulp-webpack/index.js`:
```javascript
 62: var stream = through(function(file) {
     ⋮
 73: entry.push(file.path);
     ⋮
 90: options.entry = options.entry || entry;
     ⋮
100: var compiler = webpack(options, function(err, status) {
```

It appears the workaround would be that you'll have to separate all the processing before webpack you would normally handle within gulp into a custom [webpack loader](https://webpack.js.org/contribute/writing-a-loader/) script:
```javascript
gulp.src('src/app.js')
  .pipe(named())
  .pipe(replaceString('@env@', 'dev'))
  .pipe(webpack())
  .pipe(gulp.dest('docs/'))
```
becomes:
```javascript
gulp.src('src/app.js')
  .pipe(named())
  .pipe(webpack({
    module:{
      rules:[
        {
          test:/.js$/,
          use:[
            {
              loader: path.resolve('loaders/replaceString.js'),
              options: { /* … */ }
            }
          ]
        }
      ]
    }
  })
  .pipe(gulp.dest('docs/'))
```

I haven't run this, but that appears to be the necessary way to do this.

Or we can try to get webpack to accept a file stream in place of a file name to load on its own…

**Comment by YodasWs** [comment]
Or, better yet, you'll have to run two tasks in series:
```javascript
gulp.task('build', gulp.series(
  ()=>{
    return gulp.src('src/app.js')
      .pipe(replaceString())
      .pipe(gulp.dest('build/'))
  },
  ()=>{
    return gulp.src('build/app.js')
      .pipe(named())
      .pipe(webpack())
      .pipe(gulp.dest('docs/'))
  }
);
```

**Comment by PsychodelEKS** [comment]
The issue is still here, I try to replace string literals with gulp-inject-string on the src collection (which contains entry point), prior to webpack-stream invocation, but when it is called, the entry point file is read from fs and the changes are ignored. Running replace over minified sources breaks source maps.

```
        "@babel/core":              "^7.4.0",
        "@babel/preset-env":        "^7.4.2",
        "babel-loader":             "^8.0.5",
        "gulp":                     "^4.0.0",
        "gulp-babel":               "^8.0.0",
        "gulp-inject-string":       "^1.1.2",
        "webpack":                  "^4.29.6",
        "webpack-stream":           "^5.2.1"
```

**Comment by Janff** [comment]
hope to fix this problem. I have the same problem using gulp-header

**Comment by damianomagrini** [comment]
In the meanwhile, I've made my own plugin that actually uses the file system that was passed into it instead of `fs`. You can see it [here](https://www.github.com/damianomagrini/gulp-webpack-bundler).
